src/main/java/org/folio/service/migration/RequestSearchFieldsMigrationService.java

```diff
@@ -0,0 +1,177 @@
+
+  private static final String MIGRATION_MODULE_VERSION = "mod-circulation-storage-16.1.0";
+  private static final String ITEMS_STORAGE_URL = "/item-storage/items";
+  private static final String SERVICE_POINT_URL = "/service-points";
+  private static final String REQUEST_TABLE = "request";
+
+  public RequestSearchFieldsMigrationService(TenantAttributes attributes, Context context,
+    Map<String, String> okapiHeaders) {
+
+    super(attributes, context, okapiHeaders, REQUEST_TABLE, MIGRATION_MODULE_VERSION);
+  }
+
+  public Future<Void> processBatch(Batch<RequestSearchMigrationContext> batch) {
+    log.info("processBatch:: {} processing started", batch);
+
+    return succeededFuture(batch)
+      .compose(this::fetchRequests)
+      .compose(this::findServicePointNames)
+      .compose(this::findCallNumbers)
+      .onSuccess(this::buildNewRequests)
+      .compose(this::updateRequests)
+      .onSuccess(r -> log.info("{} processing finished successfully", batch))
+      .recover(t -> handleError(batch, t));
+  }
+
+  @Override
+  RequestSearchMigrationContext buildContext(JsonObject request) {
+    return new RequestSearchMigrationContext(request);
+  }
+
+  private Future<Batch<RequestSearchMigrationContext>> findServicePointNames(
+    Batch<RequestSearchMigrationContext> batch) {
+
+    log.info("findServicePointNames:: batch={}", batch);
+
+    Set<String> servicePointIds = batch.getRequestMigrationContexts()
+      .stream()
+      .map(RequestSearchMigrationContext::getPickupServicePointId)
+      .filter(Objects::nonNull)
+      .collect(toSet());
+
+    if (servicePointIds.isEmpty()) {
+      log.info("findServicePointNames:: 0 service points found for batch {}", batch);
+      return succeededFuture(batch);
+    }
+
+    return okapiClient.get(SERVICE_POINT_URL, servicePointIds, "servicepoints", ServicePoint.class)
+      .onSuccess(servicePoints -> saveServicePointNames(batch, servicePoints))
+      .map(batch);
+  }
+
+  private static void saveServicePointNames(Batch<RequestSearchMigrationContext> batch,
+    Collection<ServicePoint> servicePoints) {
+
+    log.info("saveServicePointNames:: batch={}, servicePoints=Collection({} elements)", batch,
+      servicePoints.size());
+
+    Map<String, String> servicePointIdToName = servicePoints.stream()
+      .collect(toMap(ServicePoint::getId, ServicePoint::getName, (a, b) -> a));
+
+    batch.getRequestMigrationContexts().forEach(ctx ->
+      ctx.setPickupServicePointName(servicePointIdToName.get(ctx.getPickupServicePointId())));
+  }
+
+  private Future<Batch<RequestSearchMigrationContext>> findCallNumbers(
+    Batch<RequestSearchMigrationContext> batch) {
+
+    log.info("findCallNumbers:: batch={}", batch);
+
+    Set<String> itemIds = batch.getRequestMigrationContexts()
+      .stream()
+      .map(RequestSearchMigrationContext::getItemId)
+      .filter(Objects::nonNull)
+      .collect(toSet());
+
+    if (itemIds.isEmpty()) {
+      return succeededFuture(batch);
```

Comment:
add logging like for service points above?

Reply:
added